smart-contract-verifier/src/vyper/compiler.rs

```diff
@@ -0,0 +1,544 @@
+        let string = match self {
+            EvmVersion::Byzantium => "byzantium",
+            EvmVersion::Constantinople => "constantinople",
+            EvmVersion::Petersburg => "petersburg",
+            EvmVersion::Istanbul => "istanbul",
+        };
+        write!(f, "{}", string)
+    }
+}
+
+impl FromStr for EvmVersion {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "byzantium" => Ok(EvmVersion::Byzantium),
+            "constantinople" => Ok(EvmVersion::Constantinople),
+            "petersburg" => Ok(EvmVersion::Petersburg),
+            "istanbul" => Ok(EvmVersion::Istanbul),
+            s => Err(format!("Unknown evm version: {}", s)),
+        }
+    }
+}
+
+// #[derive(Debug, Clone, Eq, PartialEq, Default, Serialize, Deserialize)]
+// #[serde(transparent)]
+// pub struct OutputSelection(pub BTreeMap<String, Vec<String>>);
+//
+// impl OutputSelection {
+//     /// Select all possible compiler outputs: "outputSelection: { "*": ["*"] }"
+//     /// Note that this might slow down the compilation process needlessly.
+//     pub fn complete_output_selection() -> Self {
+//         BTreeMap::from([(
+//             "*".to_string(),
+//             vec!["*".to_string()]
+//         )])
+//             .into()
+//     }
+//
+//     /// Default output selection for compiler output:
+//     ///
+//     /// `{ "*": ["abi","evm.bytecode","evm.deployedBytecode","evm.methodIdentifiers"] }`
+//     pub fn default_output_selection() -> Self {
+//         BTreeMap::from([(
+//             "*".to_string(),
+//             vec![
+//                 "abi".to_string(),
+//                 "evm.bytecode".to_string(),
+//                 "evm.deployedBytecode".to_string(),
+//                 "evm.methodIdentifiers".to_string(),
+//             ],
+//         )]).into()
+//     }
+// }
+//
+// /// TODO: test empty array
+//
+// impl AsRef<BTreeMap<String, Vec<String>>> for OutputSelection {
+//     fn as_ref(&self) -> &BTreeMap<String, Vec<String>> {
+//         &self.0
+//     }
+// }
+//
+// impl AsMut<BTreeMap<String, Vec<String>>> for OutputSelection {
+//     fn as_mut(&mut self) -> &mut BTreeMap<String, Vec<String>> {
+//         &mut self.0
+//     }
+// }
+//
+// impl From<BTreeMap<String, Vec<String>>> for OutputSelection {
+//     fn from(s: BTreeMap<String, Vec<String>>) -> Self {
+//         OutputSelection(s)
+//     }
+// }
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Settings {
+    #[serde(
+    default,
+    skip_serializing_if = "Option::is_none")]
+    pub evm_version: Option<EvmVersion>,
+    pub optimize: bool,
+    #[serde(default)]
+    pub output_selection: OutputSelection,
+}
+
+impl Settings {
+    /// Creates a new `Settings` instance with the given `output_selection`
+    pub fn new(output_selection: impl Into<OutputSelection>) -> Self {
```

Comment:
you can just accept `OutputSelection`, and call `.into()` when you're calling the function, like
```Settings::new(foo.into())```